### PR TITLE
Fixes bug 1221957 - Added a Product summary.

### PR DESCRIPTION
**Pull request by AdrianGaudebert**
@peterbe r?

### CODE REVIEW

**Comment by peterbe** [comment]
The numbers are VERY different
screenshot 2015-11-11 16 53 43

screenshot 2015-11-11 16 54 26

**Comment by peterbe** [comment]
Also, what's up with it being 99.3% only on Firefox 38.4.0esr?!

File: webapp-django/crashstats/signature/views.py [hunk 1]
```diff
@@ -439,6 +439,32 @@ def signature_summary(request, params):
...
+
+    product_version_facet = None
+
+    # If there is a version or a product filter, we need to make a separate
+    # query so that we can show all versions and not just the one asked for.
+    if (
+        ('version' in params and params['version'])
+        or ('product' in params and params['product'])
+    ):
+        params_copy = dict(params)
```
Comment (peterbe) [inline webapp-django/crashstats/signature/views.py]: that makes a shallow copy. Is that OK? I.e. are there no deeper levels that 1 level?

File: webapp-django/crashstats/signature/views.py [hunk 2]
```diff
@@ -439,6 +439,32 @@ def signature_summary(request, params):
...
+        del params_copy['product']
+        del params_copy['version']
+        params_copy['_aggs.product'] = ['version']
+
+        try:
+            search_results = api.get(**params_copy)
+        except models.BadStatusCodeError as e:
```
Comment (peterbe) [inline webapp-django/crashstats/signature/views.py]: yeah! "as e"

**Comment by peterbe** [comment]
Now's maybe not the time but that page could really do with some different grid system. When one table gets really large it does get quite messed up. 
screenshot 2015-11-11 16 57 47

**Comment by AdrianGaudebert** [comment]
Stage postgres and ES are very inconsistent. I wouldn't give any credit to those numbers (though if the same discrepancy happens on prod, we'll have a problem). 

Regarding the boxes, it's definitely out of scope, but I agree with you. Ideally we should use some sort of flex system to make all of that better.

File: webapp-django/crashstats/signature/views.py (hunk 1 above)
Comment (AdrianGaudebert) [inline webapp-django/crashstats/signature/views.py]: No, I'm not changing the content of the values, just removing some keys, so that's fine.

**Comment by AdrianGaudebert** [comment]
Actually, after talking to KaiRo on IRC, I'm going to revisit this a bit to remove all filters to the query instead of just product and version.

**Comment by peterbe** [comment]
By the way, I can't find anything wrong with the code. So if you're confident the data queries are right and that the results will be more sane on prod, then r+

**Comment by AdrianGaudebert** [comment]
@peterbe I just added a new commit, r?

**Comment by peterbe** [comment]
Otherwise, r+

**Pull request merged**